### PR TITLE
Add Easyrpg savegame chunk

### PR DESCRIPTION
**Pull request by Ghabry**
This adds a new EasyRpg chunk to the savegame with one field: version.

I picked 0xc8 (200) because Cherry asked us to look at the decimals and RPG_RT starts at 0x64 (100)

Patch for the Player site, wrapping our compatibility hack in a if-block.

```
diff --git a/src/player.cpp b/src/player.cpp
index 64e3ea35..e833188a 100644
--- a/src/player.cpp
+++ b/src/player.cpp
@@ -812,11 +812,12 @@ static void OnMapSaveFileReady(FileRequestResult*) {
 	// Compatibility hack for pre 0.6 EasyRPG Player saves.
 	// Old savegames accidentally wrote animation_type as
 	// continuous for all events.
-	Main_Data::game_player->SetAnimationType(Game_Character::AnimType::AnimType_non_continuous);
-	Game_Map::GetVehicle(Game_Vehicle::Boat)->SetAnimationType(Game_Character::AnimType::AnimType_non_continuous);
-	Game_Map::GetVehicle(Game_Vehicle::Ship)->SetAnimationType(Game_Character::AnimType::AnimType_non_continuous);
-	Game_Map::GetVehicle(Game_Vehicle::Airship)->SetAnimationType(Game_Character::AnimType::AnimType_non_continuous);
-
+	if (Main_Data::game_data.easyrpg_data.version == RPG::SaveEasyRpgData::EasyRpg_Version_legacy) {
+		Main_Data::game_player->SetAnimationType(Game_Character::AnimType::AnimType_non_continuous);
+		Game_Map::GetVehicle(Game_Vehicle::Boat)->SetAnimationType(Game_Character::AnimType::AnimType_non_continuous);
+		Game_Map::GetVehicle(Game_Vehicle::Ship)->SetAnimationType(Game_Character::AnimType::AnimType_non_continuous);
+		Game_Map::GetVehicle(Game_Vehicle::Airship)->SetAnimationType(Game_Character::AnimType::AnimType_non_continuous);
+	}
 	Main_Data::game_player->Refresh();
 
 	RPG::Music current_music = Main_Data::game_data.system.current_music;
```

### CODE REVIEW

File: src/generated/rpg_saveeasyrpgdata.h [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+
+/**
+ * RPG::SaveEasyRpgData class.
+ */
+namespace RPG {
+	class SaveEasyRpgData {
+	public:
+		enum EasyRpg_Version {
+			EasyRpg_Version_legacy = 0,
+			EasyRpg_Version_060 = 1
```
Comment (fmatthew5876) [inline src/generated/rpg_saveeasyrpgdata.h]: I'm not sure this is the right place to encode this mapping. Essentially we're creating a dependency on Player versions from liblcf.

I would probably just make it an int at the liblcf level, with a constant for the 0 legacy value. The mapping of chunk integer values to easyrpg versions could be done in Player.

Comment (Ghabry) [reply src/generated/rpg_saveeasyrpgdata.h]: makes sense.
But that PrepareSave sets the version field to 1 is okay?

Comment (fmatthew5876) [reply src/generated/rpg_saveeasyrpgdata.h]: Lets add a version argument to prepare save that gets passed in by player. Hardcoding 1 into that method is basically no different than this enum.

Comment (fmatthew5876) [reply src/generated/rpg_saveeasyrpgdata.h]: Also, the value we put in could just directly map to the release version:
0.6.0 -> 600
0.10.12 -> 1012
1.0 -> 10000

Then its pretty obvious from inspection and we don't need an extra mapping.

File: generator/csv/fields_easyrpg.csv [hunk 2]
```diff
@@ -0,0 +1,3 @@
+#Structure,Field,Size Field?,Type,Index,Default Value,PersistIfDefault,Is2k3,Comment
+Save,easyrpg_data,f,SaveEasyRpgData,0xc8,,0,0,Additional save data written by EasyRPG Player
+SaveEasyRpgData,version,,Enum<EasyRpg_Version>,1,0,0,0,Savegame version
```
Comment (carstene1ns) [inline generator/csv/fields_easyrpg.csv]: Is this `enum` a leftover? Seems unused.

Comment (Ghabry) [reply generator/csv/fields_easyrpg.csv]: I don't even know how I got this commited, I deleted the file before o.O

Comment (Ghabry) [reply generator/csv/fields_easyrpg.csv]: oh this is in fields, not enum. good find!

**Pull request merged**